src/main/java/com/artipie/rpm/meta/XmlRepomd.java

```diff
@@ -126,25 +133,37 @@ public void close() throws IOException {
         try {
             this.xml.writeEndElement();
             this.xml.close();
+            this.stream.close();
         } catch (final XMLStreamException err) {
             throw new IOException("Failed to close", err);
         }
     }
 
     /**
-     * New XML stream writer from path.
+     * New stream from path.
      * @param path File path
+     * @return Output stream
+     */
+    private static OutputStream outputStream(final Path path) {
+        try {
+            return Files.newOutputStream(path);
+        } catch (final IOException err) {
+            throw new UncheckedIOException("Failed to open file stream", err);
+        }
+    }
+
+    /**
+     * New XML stream writer from path.
+     * @param stream Output steam
      * @return XML stream writer
      */
-    private static XMLStreamWriter xmlStreamWriter(final Path path) {
+    private static XMLStreamWriter xmlStreamWriter(final OutputStream stream) {
```

Comment:
@olenagerasimova this method and previous method `outputStream` will no longer be needed